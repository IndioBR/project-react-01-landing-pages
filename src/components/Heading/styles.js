import styled from 'styled-components';
import { css } from 'styled-components';
import { theme } from '../../styles/theme';

export const Title = styled.h1`
  ${({ theme, light }) => css`
    color: ${light ? theme.colors.white : theme.colors.primaryColor};
  `}
`;
