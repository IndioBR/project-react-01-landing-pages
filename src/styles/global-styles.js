import { css } from 'styled-components';
import { createGlobalStyle } from 'styled-components';
import { theme } from './theme';

export const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
  }

  body {
    ${() => css`
      background: ${theme.colors.primaryBg};
    `}
  }
`;
