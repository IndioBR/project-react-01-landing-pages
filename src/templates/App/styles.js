import styled, { css } from 'styled-components';
import { theme } from '../../styles/theme';

export const Wrapper = styled.div`
  ${() => css`
    background: ${theme.colors.secondaryBg};
  `}
`;
