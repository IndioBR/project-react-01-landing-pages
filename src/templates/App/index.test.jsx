import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import Home from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

test('renders learn react link', () => {
  const { debug } = renderTheme(<Home />);
  debug();
  const headingContainer = screen.getByRole('heading', {
    name: 'hello',
  }).parentElement;
  expect(headingContainer).toHaveStyle({
    background: theme.colors.secondaryBg,
  });
  //OU
  //expect(headingContainer).toMatchSnapshot() --- Para testar estilo
  //OU
  //expect(headingContainer).toHaveStyleRule('background', 'red') --- Para testar um estilo espec√≠fico
});
