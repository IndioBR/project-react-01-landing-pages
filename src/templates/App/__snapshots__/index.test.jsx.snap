// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  class="sc-bczRLJ gOqAgo"
>
  <h1>
    hello
  </h1>
</div>
`;
